core/helper.go

```diff
@@ -103,118 +104,135 @@ func ParseResponse[T Response](response *http.Response, value *T) error {
 	return nil
 }
 
-// Deprecated: deprecated in version 1.4.0. Please use SendGet2.
+// Deprecated: deprecated in v1.4.0. Please use SendGet2.
 func sendGet[T Response](requestBuilder *RequestBuilder, params interface{}, errorMapping ErrorMapping, value *T) error {
 	requestInfo, err := requestBuilder.ToGetRequestInformation(params)
 	if err != nil {
 		return err
 	}
 
-	response, err := requestBuilder.Client.Send(requestInfo, errorMapping)
+	response, err := requestBuilder.Client.Send(context.Background(), requestInfo, errorMapping)
 	if err != nil {
 		return err
 	}
 
 	return ParseResponse(response, value)
 }
 
-func SendGet2(requestBuilder *RequestBuilder, config *RequestConfiguration) error {
+func SendGet2(ctx context.Context, requestBuilder *RequestBuilder, config *RequestConfiguration) error {
 	requestInfo, err := requestBuilder.ToGetRequestInformation2(config)
 	if err != nil {
 		return err
 	}
 
-	response, err := requestBuilder.Client.Send(requestInfo, config.ErrorMapping)
+	response, err := requestBuilder.Client.Send(ctx, requestInfo, config.ErrorMapping)
 	if err != nil {
 		return err
 	}
 
 	return ParseResponse(response, &config.Response)
 }
 
-// Deprecated: deprecated in version 1.4.0. Please use SendPost2.
+// Deprecated: deprecated in v1.4.0. Use `SendPost2` instead.
 func sendPost[T Response](requestBuilder *RequestBuilder, data interface{}, params interface{}, errorMapping ErrorMapping, value *T) error {
 	requestInfo, err := requestBuilder.ToPostRequestInformation2(data, params)
 	if err != nil {
 		return err
 	}
 
-	response, err := requestBuilder.Client.Send(requestInfo, errorMapping)
+	response, err := requestBuilder.Client.Send(context.Background(), requestInfo, errorMapping)
 	if err != nil {
 		return err
 	}
 
 	return ParseResponse(response, value)
 }
 
-func SendPost2(requestBuilder *RequestBuilder, config *RequestConfiguration) error {
+func SendPost2(ctx context.Context, requestBuilder *RequestBuilder, config *RequestConfiguration) error {
```

Comment:
Change in signature violates Semantic Versioning for a minor release, see the [introduction](https://semver.org/#introduction), creating an additional method (ex: SendPost3, or SendPostWithContext) would be suitable.